src/arguments/utility.py

```diff
@@ -33,13 +34,123 @@
 
 console = Console()
 
+class Playbook():
+    def __init__(self):
+        self.linux_path = "/home/{}/Documents/dynamic".format(os.getenv('USER'))
+        self.mac_path = "/Users/{}/Documents/dynamic".format(os.getenv('USER'))
+        self.file_name = 'dynamic_playbook.json'
+        self.key = 'DYNAMIC'
+
+    @property
+    def playbook_path(self):
+        # Create an environment variable 'DYNAMIC' containing the path of dynamic_playbook.json and returns it
+        if not os.getenv(self.key):
+            if(sys.platform=='linux'):
+                os.environ[self.key] = os.path.join(self.linux_path, self.file_name)
+            elif(sys.platform=='darwin'):
+                os.environ[self.key] = os.path.join(self.mac_path, self.file_name)
+        return os.getenv(self.key)
+
+    @property
+    def playbook_template(self):
+        # Basic template and fields of playbook
+        return  {"time_of_update": time.time(),"items_stackoverflow":[]}
+
+    @property
+    def playbook_content(self):
+        # Reads playbook data from local storage and returns it
+        try:
+            with open(self.playbook_path, 'r') as playbook:
+                return json.load(playbook)
+        except FileNotFoundError:
+            os.makedirs(os.path.dirname(self.playbook_path), exist_ok=True)
+            with open(self.playbook_path, 'w') as playbook:
+                json.dump(self.playbook_template, playbook, ensure_ascii=False)
+            return self.playbook_content
+
+    @playbook_content.setter
+    def playbook_content(self, value):
+        if isinstance(value, dict):
+            with open(self.playbook_path, 'w') as playbook:
+                json.dump(value, playbook, ensure_ascii=False)
+        else:
+            raise ValueError("value should be of type dict")
+
+    def is_question_in_playbook(self, question_id):
+        content = self.playbook_content
+        for entry in content['items_stackoverflow']:
+            if int(entry['question_id']) == int(question_id):
+                return True
+        return False
+
+    def add_to_playbook(self, stackoverflow_object, question_id):
+        """
+        Receives a QuestionsPanelStackoverflow object and
+        saves data of a particular question into playbook
+        Saves playbook in the following format
+        {
+            time_of_update: unix,
+            items_stackoverflow:
+            [
+                {
+                    time: unix timestamp
+                    question_id: 123456,
+                    question_title: 'question_title',
+                    question_link:  'link',
+                    answer_body: 'body of the answer'
+                },
+                ...
+            ]
+        """
+        if self.is_question_in_playbook(question_id):
+            console.print("[red] Question is already in the playbook, No need to add")
```

Comment:
Just a doubt. Do we save every time the user asks the question 💭 ? If yes, we will have to remove this condition because users can ask the same question many times. If no, please let me know.

Please keep me informed 🙂

Reply:
No, I think if the same question were asked to it will pop up an error that it already exists. @NamamiShanker Can confirm from his side as well.

Reply:
> No, I think if the same question is asked to it will pop up an error that it already exists.
But, the users can ask the same question more than a time

Reply:
When the user searches for a question, he gets the results from StackOverflow. He then has an option to save that question in playbook as a note or reminder using the `p` key. These questions will be stored and visible in his playbook which is accessible by `-p` flag.

User can search any questions he want any number of times, but if he tries to store a question which is already stored in the playbook, it will show `Question already in the playbook, no need to save`

Reply:
Thanks for the explanation @namamishanker

Reply:
Thanks for the explanation @namamishanker